### PR TITLE
glslang: Enable optimization support

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: mingw-w64-glslang/PKGBUILD [hunk 1]
```diff
@@ -19,16 +19,19 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-python")
 source=(${_realname}-${pkgver}.tar.gz::https://github.com/KhronosGroup/glslang/archive/${pkgver}.tar.gz
         001-respect-destdir-for-compat-symlink.patch::https://github.com/KhronosGroup/glslang/commit/171a3220.patch
-        002-remove-deprecated-mingw-secure-api.patch)
+        002-remove-deprecated-mingw-secure-api.patch
+        https://github.com/KhronosGroup/glslang/commit/ffd07992f1b1f9e8a609338973c5beaa53d7bfa9.patch)
```
Comment (Biswa96) [inline mingw-w64-glslang/PKGBUILD:23]: This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository. If the patch is not in main glslang repository, you can add the patch file in this repository. And add the PR link as a comment above `patch` command.

**Comment by Biswa96** [comment]
Some questions.
1. Should glslang be linked with packaged version of spirv-tools? The glslang repository uses different version of spirv-tools than ours.
2. Should we wait for that patch to be merged in upstream?
3. Does this require to re-build dependent packages?

File: mingw-w64-glslang/PKGBUILD (hunk 1 above)
Comment (kasper93) [reply mingw-w64-glslang/PKGBUILD:23]: Ok, let's wait a bit, it should be merged soon https://github.com/KhronosGroup/glslang/pull/3152

**Comment by kasper93** [comment]
> Should glslang be linked with packaged version of spirv-tools? The glslang repository uses different version of spirv-tools than ours.

So for full compatibility both packages in MSYS2 should have compatible versions. Which are tracked with https://github.com/KhronosGroup/glslang/blob/main/known_good.json

I tested the current versions works correctly. But indeed the `known_good.json` is something to keep in mind when making spit packages.

> Should glslang be linked with packaged version of spirv-tools?

Yes, especially that we provide static libraries too.

**Comment by kasper93** [comment]
> Should glslang be linked with packaged version of spirv-tools? The glslang repository uses different version of spirv-tools than ours.

So for full compatibility both packages in MSYS2 should have compatible versions. Which are tracked with https://github.com/KhronosGroup/glslang/blob/main/known_good.json

I tested the current versions works correctly. But indeed the `known_good.json` is something to keep in mind when making spit packages.

> Should glslang be linked with packaged version of spirv-tools?

Yes, especially that we provide static libraries too.

> Should we wait for that patch to be merged in upstream?

Yes, sure let's do that.

> Does this require to re-build dependent packages?

Yes, if they are linking to static version. for shared one shouldn't be a difference. But I rebuild all just in case.

**Comment by kasper93** [comment]
> Should we wait for that patch to be merged in upstream?

Merged now, updated with the version of a patch from main branch.

**Comment by kasper93** [comment]
```
   D:/M/msys64/ucrt64/include/c++/13.2.0/cstdlib:79:15: fatal error: stdlib.h: No such file or directory
     79 | #include_next <stdlib.h>
        |               ^~~~~~~~~~
  compilation terminated.
```
huh?

**Comment by lazka** [comment]
> huh?

#13608 - a retry usually fixes it

**Comment by kasper93** [comment]
> #13608 - a retry usually fixes it

I see. I needed to update vulkan-validation-layers, because build failed with vulkan-headers from the repo.

**Comment by lazka** [comment]
you also need to update the checksum.

**Comment by lazka** [comment]
hm, it was revered https://github.com/KhronosGroup/Vulkan-ValidationLayers/commit/e6bdb8d71409a96a4174589ea195d0dc1e920625

**Comment by kasper93** [comment]
> hm, it was revered

Yes, I see that, but it still doesn't compile. There are a lot of dependencies between those vulkan packages. I'm trying to unwind what can be done here.

**Comment by lazka** [comment]
with the revert I get lots of other errors, it seems like we need to update SPIRV-Tools also maybe

**Comment by kasper93** [comment]
> good luck, I'm out :)

For now the only way is to revert vulkan headers update and wait for approprieate spirv-tools tag.

https://github.com/msys2/MINGW-packages/pull/18212

**Comment by lazka** [comment]
I guess we can't get ucrt64 to pass here :/

good to merge besides that?

**Comment by kasper93** [comment]
> good to merge besides that?

good on my end.

**Pull request merged**